test/acceptance/features/steps/generic_testapp.py

```diff
@@ -19,16 +20,58 @@ def get_env_var_value(self, name):
         else:
             return None
 
+    def check_for_404_env_var_value(self, name):
+        resp = polling2.poll(lambda: requests.get(url=f"http://{self.route_url}/env/{name}"),
+                             check_success=lambda r: r.status_code == 404, step=5, timeout=400)
+        print(f'env endpoint response code: {resp.status_code}')
+        if resp.status_code == 404:
+            return True
+
+    def install_with_env(self):
+        create_new_app_output, exit_code = self.cmd.run(
+            f"oc new-app --docker-image={self.app_image} --name={self.name} -n {self.namespace} -e SERVICE_BINDING_ROOT={self.bindingRoot}")
+        assert exit_code == 0, f"Non-zero exit code ({exit_code}) returned when attempting to create a new app: {create_new_app_output}"
+        assert self.openshift.expose_service_route(self.name,
+                                                   self.namespace) is not None, "Unable to expose service route"
+        return self.is_running(wait=True)
+
+    def get_file_value(self, file_path):
+        resp = polling2.poll(lambda: requests.get(url=f"http://{self.route_url}{file_path}"),
+                             check_success=lambda r: r.status_code in [200, 404], step=5, timeout=400)
+        print(f'file endpoint response: {resp.text} code: {resp.status_code}')
+        if resp.status_code == 200:
+            return resp.text
+        else:
+            return None
+
 
 @given(u'Generic test application "{application_name}" is running')
 def is_running(context, application_name):
+    application = GenericTestApp(application_name, context.namespace.name)
+    if not application.is_running():
+        print("application is not running, trying to import it")
+        application.install_with_env()
+    context.application = application
+
+
+@given(u'Generic test application "{application_name}" is running without SERVICE_BINDING_ROOT')
```

Comment:
this is the same as https://github.com/redhat-developer/service-binding-operator/pull/735/files#diff-6c87266c9589bcb7eb08cba87b69f554e33f4a75ebc2f89ae3dd9060c69cce15R48 - please remove and use it instead.

Reply:
This is without `SERVICE_BINDING_ROOT`

Reply:
right, and the referred step is exactly that.

Reply:
done